### ISSUE TITLE
TypeError in sklearn.linear_model.LogisticRegression with L-BFGS solver

### ISSUE THREAD

**Issue by Aayush-N** [pyodide/pyodide#1440]
Hello everyone, 

I'm trying to run the following code from the [sklearn documentation ](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html):

```
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
X, y = load_iris(return_X_y=True)
clf = LogisticRegression(random_state=0).fit(X, y)
print(clf.predict(X[:2, :]))
print(clf.predict_proba(X[:2, :]))
print(clf.score(X, y))
```

In the browser I get this:

```
Pyodide has suffered a fatal error, refresh the page. Please report this to the Pyodide maintainers.

The cause of the fatal error was:

```

In the console I see this error:

```
Uncaught (in promise) TypeError: Cannot read property 'apply' of undefined
    at obj.<computed> (pyodide.asm.js:9)
    at mainlb_ (:8000/code-editor/hello/<anonymous>:wasm-function[90]:0x5b9c)
    at setulb_ (:8000/code-editor/hello/<anonymous>:wasm-function[89]:0x5b36)
    at byn$fpcast-emu$setulb_ (:8000/code-editor/hello/<anonymous>:wasm-function[116]:0xd547)
    at f2py_rout__lbfgsb_setulb (:8000/code-editor/hello/<anonymous>:wasm-function[69]:0x2264)
    at byn$fpcast-emu$f2py_rout__lbfgsb_setulb (:8000/code-editor/hello/<anonymous>:wasm-function[147]:0xd9d2)
    at fortran_call (:8000/code-editor/hello/<anonymous>:wasm-function[83]:0x50c7)
    at byn$fpcast-emu$fortran_call (:8000/code-editor/hello/<anonymous>:wasm-function[153]:0xda1c)
    at _PyObject_MakeTpCall (cdn.jsdelivr.net/pyodide/v0.17.0a2/full/pyodide.asm.wasm:wasm-function[766]:0x1cd1a3)
    at cdn.jsdelivr.net/pyodide/v0.17.0a2/full/pyodide.asm.wasm:wasm-function[2771]:0x2b93e9

```

I tried running it with the latest alpha release and the stable v16 and see the same error in both. 

Any help to overcome this is greatly appreciated, thanks!

**Comment by rth** [comment]
I think there is an issue with the L-BFGS scipy solver used by default in LogisticRegression. Related to https://github.com/pyodide/pyodide/issues/834.

As a workaround you can use `LogisticRegression(random_state=0, solver='liblinear')` which should work and produce an equivalent model.

**Comment by Aayush-N** [comment]
This seems to make it work, thank you so much.

**Comment by CKPOON0619** [comment]
Similar error I got with the L-BFGS Solver when I run statsmodels ARIMA function.
```
Stack (most recent call first):
pyodide.asm.js:14   File "/lib/python3.9/site-packages/scipy/optimize/lbfgsb.py", line 319 in _minimize_lbfgsb
pyodide.asm.js:14   File "/lib/python3.9/site-packages/scipy/optimize/lbfgsb.py", line 192 in fmin_l_bfgs_b
pyodide.asm.js:14   File "/lib/python3.9/site-packages/statsmodels/tsa/arima_model.py", line 566 in _fit_start_params
pyodide.asm.js:14   File "/lib/python3.9/site-packages/statsmodels/tsa/arima_model.py", line 945 in fit
pyodide.asm.js:14   File "<exec>", line 42 in <module>
pyodide.asm.js:14   File "/lib/python3.9/site-packages/_pyodide/_base.py", line 345 in run_async
pyodide.asm.js:14   File "/lib/python3.9/site-packages/_pyodide/_base.py", line 494 in eval_code_async
pyodide.asm.js:14   File "/lib/python3.9/asyncio/tasks.py", line 256 in __step
pyodide.asm.js:14   File "/lib/python3.9/asyncio/events.py", line 80 in _run
```

**Comment by hoodmane** [comment]
The error reproduces in v0.18.1 and on current main branch. Hopefully will be fixed by #2065, will test soon.

**Comment by hoodmane** [comment]
Okay, this reproduces even on #2056.

**Comment by hoodmane** [comment]
Relatedly, the following fails with `null function or function signature mismatch`:
```
from scipy.linalg import lapack
lapack.dlamch('Epsilon-Machine')
```

**Comment by hoodmane** [comment]
Somewhere we need to inject the following definition:
```C
int wdlamch_(char *x, ftnlen y){
    return dlamch_(x);
}
```

**Comment by hoodmane** [comment]
Okay, well indeed injecting the `wdlamch_` definition fixes this crash, but it doesn't fix the `lapack.dlamch` failure.

**Comment by hoodmane** [comment]
Oh, `lapack.dlamch('Epsilon-Machine')` failing is a regression on #2056.

**Comment by hoodmane** [comment]
Closed by #2065.
